### ISSUE TITLE
Need a hand?

### ISSUE THREAD

**Issue by mehcode** [zargony/atom-language-rust#119]
@zargony 

I'm working on [`ide-rust`](https://github.com/mehcode/atom-ide-rust/issues) in conjunction with the Atom IDE and Rust Dev Tools movements.

I have some pain points with `language-rust` and I was taking a look but it seems like at least one of them ( #82 ) has an open PR. 

I'd love to help you with adding syntax and improvements to the grammar here. Thoughts?

**Comment by zargony** [comment]
Any help would be much appreciated. Unfortunately I currently don't find much time to spend on improving language-rust. Most open issues are "complicated", i.e. they're not simple grammar rules problems but need a deeper look into how atom packages work (which I unfortunately don't have). If you'd like to stay for a while and help to move things ahead, I could as well add you as a collaborator. We should just make sure to discuss bigger changes openly first and to have working specs all time to prevent regressions.

#82 looks nice, but is missing some automated specs. It adds some lines to test.rs which was used for manually reviewing syntax highlighting before we had specs.

**Comment by jens1o** [comment]
> Any help would be much appreciated. Unfortunately I currently don't find much time to spend on improving language-rust. Most open issues are "complicated", i.e. they're not simple grammar rules problems but need a deeper look into how atom packages work (which I unfortunately don't have). 

Is this still the case?

**Comment by zargony** [comment]
Yes. I think I know Rust pretty well, but my knowledge of Atom’s JS internals is limited and RL keeps me busy, so I appreciate any help.

**Comment by mehcode** [comment]
`ide-rust` directly depends on this and @alexheretic now (and for quite some time) maintains it so he might have time and interest to help here

**Comment by jens1o** [comment]
That would be great, to further improve this project! :)

**Comment by alexheretic** [comment]
Yes I, and all atom rust users, am interested in improvements. I don't have a lot of free time myself at the moment though.

Interestingly it looks like they'll also be official rust support for atom when they adopt the tree-sitter parsing system: https://github.com/atom/language-rust. But who knows when that'll be.

**Comment by zargony** [comment]
Thanks. Looks like there’s a tree-sitter based grammar for Rust now.
